spec/bootstrap.js

```diff
@@ -0,0 +1,69 @@
+const CurlRequest = require('curl-request'),
+      knex = require('knex'),
+      Promise = require('es6-promise').Promise;
+
+const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+const WAIT_TIME = 5000;
+const MAX_RETRIES = 10;
+
+const PUBLIC_ANNOUNCEMENT1 = 'Please define env vars \
+    TEST_COUCH_URL(ie: http://admin:pass@localhost:5984) & \
+    TEST_PG_URL(ie: postgres://locahost:5432) to test locally.';
+
+const PUBLIC_ANNOUNCEMENT2 = 'Everything can also be tested with: \
+    docker-compose run test grunt test';
+
+const curl = new CurlRequest();
+
+const waitForDb = async ({url, fn, retries=0}) => {
+  if(retries ++ >= MAX_RETRIES) {
+    console.log(`****** ERROR: Unable to connect: ${url}`);
+    process.exit(1);
+  }
+  if(!await fn()) {
+    if(!url) {
+      console.log(PUBLIC_ANNOUNCEMENT1);
+      console.log(PUBLIC_ANNOUNCEMENT2);
+      process.exit(1);
+    }
+    console.log(`====> Waiting on ${url} to appear.`);
+    await wait(WAIT_TIME);
+    await waitForDb({url: url, fn: fn, retries: retries});
+  }
+};
+
+const waitForCouch = async (url) => {
+  await waitForDb({url: url, fn: async () => {
+    try {
+      const {statusCode} = await curl.get(url);
+      if(statusCode === 200) {
+        console.log(`- Couch [${url}] is now avaliable.`);
+        return true;
+      }
+    } catch(err) {
+      //Ignore
```

Comment:
Why would you want to ignore this?

Reply:
good catch; pun intended. need to document or remove this.